main.go

```diff
@@ -0,0 +1,231 @@
+	}
+
+	// create messages chan
+	messages := make(chan *sarama.ConsumerMessage)
+	go consumePartition(consumer, *topicPtr, int32(*partitionPtr), messages, offsetArray)
+
+	// create signals chan
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt)
+
+	consumed := 0
+ConsumerLoop:
+	for {
+		select {
+		case msg := <-messages:
+
+			// make messageBytes channel
+			messageBytes := make(chan []byte)
+
+			// if jsonOutPtr is set to 1, then output JSON and republish
+			// otherwise just republish message
+			if *jsonOutPtr == 1 {
+				go outputJSON(msg, messageBytes)
+			} else {
+				go func() {
+					messageBytes <- msg.Value
+				}()
+			}
+
+			// create producer message
+			producerMessage := &producer.Message{
+				Value: <-messageBytes,
+				Topic: *topicPtr,
+			}
+
+			//create new producer
+			p, err := producer.New(&producer.Config{Acks: &producer.WaitForAll, BrokerAddrs: []string{*brokerPtr}})
+			if err != nil {
+				fmt.Errorf("error creating producer: %v", err)
+				os.Exit(1)
+			}
+
+			// republish message
+			fmt.Printf("Republishing message for offset: %v \n", msg.Offset)
+			partition, offset, err := p.Send(producerMessage)
+			if err != nil {
+				fmt.Errorf("error republishing message with offset: %v, %v \n", msg.Offset, err)
+				os.Exit(1)
+			}
+			fmt.Printf("Successfully republished message with offset %v to topic: %v using partition: %v new offset: %v \n", msg.Offset, *topicPtr, partition, offset)
+			fmt.Println("---------------------")
+
+			consumed++
+
+			// check if we have hit the end of the array of messages
+			// if so, exit the tool by breaking the ConsumerLoop
+            if consumed == len(offsetArray) {
+                break ConsumerLoop
+            }
+		case <-signals:
+			fmt.Println("break")
+			break ConsumerLoop
+		}
+	}
+}
+
+func outputJSON(msg *sarama.ConsumerMessage, messageBytes chan []byte) {
+	schemaStruct := schemas.IdentifySchema(*topicPtr)
+	var err error
+
+	// Get schema from schema registry and use it to create avro consumer
+	fmt.Printf("Getting schema for topic: %v \n", *topicPtr)
+	schema, err := schema.Get(*schemaRegistryPtr, *schemaPtr)
+	if err != nil {
+		fmt.Errorf("rror getting schema: %v", err)
+		os.Exit(1)
+	}
+	fmt.Printf("Retrieved schema for topic: %v \n", *topicPtr)
+
+	consumerAvro := &avro.Schema{
+		Definition: schema,
+	}
+
+	// Unmarshal message value and assign it to schemaStruct
+	fmt.Printf("Unmarshalling message for offset: %v \n", msg.Offset)
+	if err = consumerAvro.Unmarshal(msg.Value, schemaStruct); err != nil {
+		fmt.Errorf("error unmarshalling avro message for offset: %v, %v \n", msg.Offset, err)
+		os.Exit(1)
+	}
+	fmt.Printf("Message successfully unmarshalled for offset: %v \n", msg.Offset)
+
+	// Print the unmarshalled message out to the terminal
+	fmt.Println("Unmarshalled message:")
+	data, err := json.Marshal(schemaStruct)
+	if err != nil {
+		fmt.Errorf("error marshalling JSON message for offset: %v, %v \n", msg.Offset, err)
```

Comment:
I think this won't print the error. Perhaps use panic like you have done elsewhere.

Reply:
Sure, I will replace the error line to just output the message without the error and then add the panic below it. Like so:
`fmt.Errorf("error marshalling JSON message for offset: %v \n", msg.Offset)` 
`panic(err)`
`os.Exit(1)`

Resolved by [commit](https://github.com/companieshouse/kafka-management-tool/pull/1/commits/05095461fe44fbd53f5fbe4aa41f4e1a4be70316)